bug fixes

Torehirth opened the pull request:
Changed relative paths on a lot of JS files in the code base that were causing faults in UI behaviour.

closes #119 
closes #105 
closes #113

Torehirth committed:
in terms of sidebar - delete old code, re-enter new code. Changed relative paths on a lot of files that was causing faults in UI behaviour

HelChris commented:
@Nikita-stud I put highest priority on this, the goal is to have this fixed before the review this evening.

Nikita-stud reviewed (approved):
updated version looks good. See you made some changes

HelChris committed:
Refactor styles and HTML structure for improved layout and functionality

- Updated sidebar styles to use dialog elements with animations for better user experience.

- Refined SCSS variables for typography and layout, enhancing maintainability.
- Enhanced main styles with responsive typography and layout adjustments.
- Improved task board HTML structure for better accessibility and usability.
- Added new 'Your Week' page with a task management interface.
- Implemented export functionality for task lists in various formats.

*Hopefully fixed mergeconflicts correctly!

HelChris committed:
commiting renderSidebar.mjs
